metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -1038,7 +1038,10 @@ def gen_dataset_workunits(
 
         if table.tags:
             tag_associations = [
-                TagAssociation(tag=make_tag_urn(tag.identifier())) for tag in table.tags
+                TagAssociation(
+                    tag=make_tag_urn(self.snowflake_identifier(tag.identifier()))
+                )
+                for tag in table.tags
```

Comment:
This was generating mixed case URNs as opposed to lowercasing the urn, as is done at other places where urns for snowflake tags and other entities are generated.